client/state/data-layer/freshness-verifier/README.md

```diff
@@ -0,0 +1,45 @@
+# Data Layer freshness verifier
+
+## Background
+
+Calypso components are allowed to dispatch request actions that specify their data needs (most of the times using query components). Normally the components don't need a request to be issued they just need data that is "fresh".
+
+When a component has a data need it dispatches a request action. This leaves a problem when the component is mounted, unmounted and mounted again 2 requests are executed even if the last one was executed seconds ago. This behavior could be observable in comments, if a user shows, hide, shows the comments of a post 2 requests are generated. This happens in other cases and there are unnecessary request being dispatched, possibly wasting data-usage, battery and CPU parsing the requests.
+
+## Goals and characteristics
+
+The goal of freshness verifier is to provide a way for data-layer users/handlers to specify that a given request should be ignored if an equal request was executed recently.
+Data-layer should provide a default amount of time for the request be considered fresh, but there are situations where we require data to be more recent than in other uses of the same data, so the requester should also be able to configure the freshness value.
+
+## API
+The handler when using dispatchRequest needs to pass in the options object a freshness property, that represents the default maximum amount of time passed since last execution for the request be considered fresh (and ignored).
+e.g:
+```js
+[ COMMENTS_REQUEST ]: [ dispatchRequest( fetchPostComments, addComments, announceFailure, noop, { freshness: 10000 } ) ],
+```
+The freshness value the data-layer sets acts as a default value if a data requester has a particular need for a different freshness value it can pass that value in the action being dispatched:
+```js
+{
+ type: COMMENTS_REQUEST,
+ siteId: 12886750,
+ postId: 97,
+ ...
+ freshness: 2000
+}
+```
+When a request is still fresh it does not trigger a network request and is ignored.
+
+The freshness mechanism is optional and if no freshness is set in the handler or the action the freshness mechanism is not executed. If a freshness value is set on the handler but in a given component freshness should not be used it's possible to opt-out its usage by setting `freshness: 0` in the action being dispatched. Opt-out of the mechanism is not recommended unless in the presence of a specific case where fresh data is really needed.
+
+The default value for the freshness of the actions of a given type is set in the handler so if the action does not set its own freshness value the one set in the handler of the request is used. There is no global default value if the both handler and action don't set a freshness value, freshness is not used.
```

Comment:
This paragraph might make more sense in an ordered list of where we get the values…

```md
The freshness for a specific request action is determined in the following priority:
  1. from action metadata at `meta.dataLayer.freshness`
  2. from the optional `freshness` parameter in the call to `dispatchRequest()` in the data layer handler for the action type
  3. a globally-assumed default of `0`, implying that requests will be instantly fetched as often as they dispatch
```